@xen-orchestra/web/src/pages/pool/[id]/hosts.vue

```diff
@@ -1,99 +1,59 @@
-      <UiTablePagination v-if="areHostsReady" v-bind="paginationBindings" />
-    </div>
-  </UiCard>
+  <div class="hosts" :class="{ mobile: uiStore.isMobile }">
+    <UiCard class="container">
+      <HostsTable :hosts :host-ready="areHostsReady" :has-error="hasHostFetchError" />
+    </UiCard>
+    <HostsSidePanel v-if="selectedHost" :host="selectedHost" @close="selectedHost = undefined" />
+    <UiPanel v-else-if="!uiStore.isMobile">
+      <VtsStateHero format="panel" type="no-selection" size="medium">
+        {{ t('select-to-see-details') }}
+      </VtsStateHero>
+    </UiPanel>
+  </div>
 </template>
 
 <script lang="ts" setup>
+import HostsTable from '@/components/hosts/HostsTable.vue'
+import HostsSidePanel from '@/components/hosts/panel/HostsSidePanel.vue'
 import { useXoHostCollection } from '@/remote-resources/use-xo-host-collection.ts'
-import type { HostState } from '@core/types/object-icon.type'
-import VtsCellObject from '@core/components/cell-object/VtsCellObject.vue'
-import VtsCellText from '@core/components/cell-text/VtsCellText.vue'
-import VtsObjectIcon from '@core/components/object-icon/VtsObjectIcon.vue'
 import VtsStateHero from '@core/components/state-hero/VtsStateHero.vue'
-import ColumnTitle from '@core/components/table/ColumnTitle.vue'
-import VtsTable from '@core/components/table/VtsTable.vue'
 import UiCard from '@core/components/ui/card/UiCard.vue'
-import UiObjectLink from '@core/components/ui/object-link/UiObjectLink.vue'
-import UiTablePagination from '@core/components/ui/table-pagination/UiTablePagination.vue'
-import { usePagination } from '@core/composables/pagination.composable'
-import { defineTree } from '@core/composables/tree/define-tree'
-import { useTree } from '@core/composables/tree.composable'
-import type { XoPool } from '@vates/types'
+import UiPanel from '@core/components/ui/panel/UiPanel.vue'
+import { useRouteQuery } from '@core/composables/route-query.composable'
+import { useUiStore } from '@core/stores/ui.store'
+import type { XoHost, XoPool } from '@vates/types'
 import { computed } from 'vue'
 import { useI18n } from 'vue-i18n'
 
-const props = defineProps<{
+const { pool } = defineProps<{
   pool: XoPool
 }>()
 
+const uiStore = useUiStore()
+
 const { t } = useI18n()
 
-const { areHostsReady, hostsByPool } = useXoHostCollection()
+const { areHostsReady, hostsByPool, hasHostFetchError } = useXoHostCollection()
 
-const definitions = computed(() =>
-  defineTree(hostsByPool.value.get(props.pool.id) ?? [], {
-    getLabel: 'name_label',
-  })
-)
+const hosts = computed(() => hostsByPool.value.get(pool.id) ?? [])
 
-const { nodes: hosts } = useTree(definitions)
-const { pageRecords: hostsRecords, paginationBindings } = usePagination('hosts', hosts)
+const selectedHost = useRouteQuery<XoHost | undefined>('id', {
+  toData: id => hosts.value.find(host => host.id === id),
+  toQuery: vm => vm?.id ?? '',
```

Comment:
```suggestion
  toQuery: host => host?.id ?? '',
```
Be carefull when you copy-paste